packages/redux-send-postMessage-thunk/src/sendPostMessageMiddleware.js

```diff
@@ -0,0 +1,8 @@
+
+export const createSendPostMessageMiddleware = (windowSelector, actionsMap) => ({ dispatch }) => next => (action) => {
+  const { targetWindow, targetWindowOrigin } = windowSelector();
+  if (targetWindow && actionsMap.has(action.type)) {
+    targetWindow.postMessage(action, targetWindowOrigin);
+  }
+  return next(action);
+};
```

Comment:
We also probably want to  convert the actionsMap from an array to a Set.  Nobody likes dealing with  Sets in JavaScript, and it's kind of an implementation detail that gets us fast `.has`  instead of using `array.includes`.

Maybe something like this?
```
export const createSendPostMessageMiddleware = (windowSelector, actionTypes = []) => {
  const actionTypesSet = new Set(actionTypes);
  
  return  ({ dispatch }) => next => (action) => {
  if (actionTypesSet.has(action.type)) {
     const { targetWindow, targetWindowOrigin } = windowSelector();
     if (targetWindow) {
         targetWindow.postMessage(action, targetWindowOrigin);
    }
  }
  return next(action);
}
};
```

Reply:
Updated.

Reply:
Yeah, when I thought about this again, I can't decide if we should take in the list of types as a `Set` or an `Array`.  The difference would be that if we took it as a `Set`, then it would be mutable, and a caller could change the elements that were in the `Set` after the fact.  I can't decide if that's good or bad, so we'll go with the `Array` for now, but I might choose to revisit this in the future, and possibly allow that argument to be either an `Array` or a `Set`.

Reply:
Noted. Also updated createSendPostMessageMiddleware to be the default export.